lib/pbench/server/database/models/server_config.py

```diff
@@ -172,29 +170,24 @@ def validate_server_banner(key: str, value: JSONVALUE) -> JSONVALUE:
     return value
 
 
-class ServerConfigOptions(Enum):
-    MAXIMUM_DATASET_LIFETIME = ("dataset-lifetime", validate_lifetime)
-    SERVER_BANNER = ("server-banner", validate_server_banner)
-    SERVER_STATE = ("server-state", validate_server_state)
-
-    def __init__(self, key: str, validator: Callable[[str, JSONVALUE], JSONVALUE]):
-        self.key = key
-        self.validator = validator
-
-    @classmethod
-    def lookup(cls, key: str) -> "ServerConfigOptions":
-        if not key:
-            raise ServerConfigMissingKey()
-        k = key.lower()
-        for x in cls:
-            if k == x.key:
-                return x
-        raise ServerConfigBadKey(k)
+OPTION_DATASET_LIFETIME = "dataset-lifetime"
+OPTION_SERVER_BANNER = "server-banner"
+OPTION_SERVER_STATE = "server-state"
 
-    @staticmethod
-    def validate(key: str, value: JSONVALUE) -> JSONVALUE:
-        option = ServerConfigOptions.lookup(key)
-        return option.validator(key, value)
+SERVER_CONFIGURATION_OPTIONS = {
+    OPTION_DATASET_LIFETIME: {
+        "validator": validate_lifetime,
+        "default": lambda: str(ServerConfig.config.max_retention_period),
```

Comment:
Oooh!  A "thunk"!!  How clever....  😀

Reply:
Ha; I would have considered this one "just a lambda" to defer the `PbenchServerConfig` access. Now, the `OPTION_SERVER_STATE` case is more of a "thunk".

Reply:
Nah...based on all my reading about Javascript and Redux, `OPTION_SERVER_STATE` is an "action creation function" (or "action creator").  On the other hand, deferring access is one of the prime reasons for creating and using thunks.  (The more obvious one for Redux is enabling asynchronous activities during store updates.)